azure-iot-device/samples/async-hub-scenarios/provision_symmetric_key_client_cert_issuance.py

```diff
@@ -0,0 +1,88 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import asyncio
+from azure.iot.device.aio import ProvisioningDeviceClient
+import os
+from azure.iot.device.aio import IoTHubDeviceClient
+from azure.iot.device import Message
+import uuid
+import logging
+from azure.iot.device import X509
+
+messages_to_send = 10
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+logging.basicConfig(level=logging.DEBUG, filename="newfile.log")
+
+
+async def main():
+    provisioning_device_client = ProvisioningDeviceClient.create_from_symmetric_key(
+        provisioning_host=provisioning_host,
+        registration_id=registration_id,
+        id_scope=id_scope,
+        symmetric_key=symmetric_key,  # authenticate for DPS
+        client_certificate_csr="some csr data",  # kwarg
+    )
+
+    registration_result = await provisioning_device_client.register()
+
+    print("The complete registration result is")
+    print(registration_result.registration_state)
+
+    print("the issued certificate")
+    # include new field called "issued_client_certificate" in the model of registration result.
+    print(registration_result.registration_state.issued_client_certificate)
+
+    with open("device_cert.pem", "w") as out_ca_pem:
+        cert_data = registration_result.registration_state.issued_client_certificate
+        # not sure if i need to do this
+        # cert_pem_data = str(base64.b64decode(cert_data), "ascii")
```

Comment:
why is this line commented?

Reply:
both these lines i have kept deliberately and can only be finalized when I run the sample